### PR TITLE
Support dep updates/hints

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: conda_forge_tick/migrators/version.py [hunk 1]
```diff
@@ -622,6 +632,22 @@ def pr_body(self, feedstock_ctx: FeedstockContext) -> str:
             )
         for p in pred:
             body += template.format(name=p[0], new_version=p[1])
+
+        update_deps = feedstock_ctx.attrs.get("conda-forge.yml", {}).get("bot", {}).get("update_deps", False)
```
Comment (beckermr) [inline conda_forge_tick/migrators/version.py:636]: We should always hint.

Comment (CJ-Wright) [reply conda_forge_tick/migrators/version.py:636]: Maybe, do you want to hint if we update directly?

File: conda_forge_tick/migrators/version.py [hunk 2]
```diff
@@ -622,6 +632,22 @@ def pr_body(self, feedstock_ctx: FeedstockContext) -> str:
...
+        if update_deps == 'hint':
+            deps = extract_deps_from_source(os.path.join(feedstock_dir, 'recipe'), self.import_cf_map)
+            dep_comparison = compare_depfinder_audit(deps, feedstock_ctx.attrs, feedstock_ctx.attrs['name'])
+            if dep_comparison:
+                df_cf = "\n-".join(self.dep_comparison["df_minus_cf"])
+                cf_df = "\n-".join(self.dep_comparison["cf_minus_df"])
+                hint = f'\n\nDependency Analysis\n-------------' \
+                       f'Analysis of the source code by depfinder seems to show a discrepency between' \
```
Comment (beckermr) [inline conda_forge_tick/migrators/version.py:644]: maybe make `depfinder` a link so ppl can find it and look at it?

Comment (CJ-Wright) [reply conda_forge_tick/migrators/version.py]: Actually @scopatz brought up the point that we might use multiple systems to provide inspection at some point, so we might want to leave this out.

File: conda_forge_tick/migrators/version.py (hunk 1 above)
Comment (beckermr) [reply conda_forge_tick/migrators/version.py]: yes. we should say we did it in the pr text

File: conda_forge_tick/migrators/version.py (hunk 2 above)
Comment (beckermr) [reply conda_forge_tick/migrators/version.py]: sgtm!

File: conda_forge_tick/migrators/version.py [hunk 3]
```diff
@@ -622,6 +623,44 @@ def pr_body(self, feedstock_ctx: FeedstockContext) -> str:
...
             body += template.format(name=p[0], new_version=p[1])
+
+        update_deps = (
+            feedstock_ctx.attrs.get("conda-forge.yml", {})
+            .get("bot", {})
+            .get("inspection", "hint")
+        )
+        if update_deps == "hint":
+            deps = extract_deps_from_source(
+                os.path.join(feedstock_ctx.feedstock_dir, "recipe"), self.import_cf_map,
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators/version.py:634]: `self.import_cf_map` needs to be implemented still

File: conda_forge_tick/migrators/version.py [hunk 4]
```diff
@@ -622,6 +631,45 @@ def pr_body(self, feedstock_ctx: FeedstockContext) -> str:
...
+            dep_comparison = compare_depfinder_audit(
+                deps, feedstock_ctx.attrs, feedstock_ctx.attrs["name"],
+            )
+            hint = f"\n\nDependency Analysis\n--------------------\n\n"
+            hint += (
+                "Please note that this analysis is **highly experimental**. "
+                "The aim here is to make maintenance easier by inspecting when dependencies have moved. "
+                "If you do not want hinting of this kind ever please add "
+                "`bot: inspection: false` to your `conda-forge.yml`. "
+                "If you encounter issues with this feature please ping the bot team `conda-forge/bot`.\n\n"
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators/version.py:653]: @ocefpaf can you take a look at these lines? I was hoping to put the disclaimer here. Although maybe we should have an independent disclaimer below for the optional deps.

File: conda_forge_tick/migrators/version.py [hunk 5]
```diff
@@ -622,6 +631,45 @@ def pr_body(self, feedstock_ctx: FeedstockContext) -> str:
...
+            )
+            if dep_comparison:
+                df_cf = "\n-".join(dep_comparison["df_minus_cf"])
+                cf_df = "\n-".join(dep_comparison["cf_minus_df"])
+                hint += (
+                    f"Analysis of the source code shows a discrepancy between"
+                    f"the library's imports and the package's stated requirements"
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators/version.py:660]:
```suggestion
                    f" the library's imports and the package's stated requirements"
```

File: conda_forge_tick/migrators/version.py [hunk 6]
```diff
@@ -622,6 +631,45 @@ def pr_body(self, feedstock_ctx: FeedstockContext) -> str:
...
+                    f" in the meta.yaml.\n"
+                    f"packages found by inspection but not in the meta.yaml\n"
+                    f"{df_cf}"
+                    f"packages found in the meta.yaml but not found by inspection\n"
+                    f"{cf_df}"
+                )
+            else:
+                hint += (
+                    "Analysis of the source code shows **no** discrepancy between"
+                    "the library's imports and the package's stated requirements in the meta.yaml."
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators/version.py:670]:
```suggestion
                    " the library's imports and the package's stated requirements in the meta.yaml."
```

**Comment by CJ-Wright** [comment]
If we don't find anything we should tell the maintainer that their recipe is in good standing, right? But they should take that with a grain of salt, since we could have missed something.

**Comment by beckermr** [comment]
Hmmmm. Ok.

**Pull request merged**